src/Input/InputParamTrait.php

```diff
@@ -10,129 +10,94 @@
 
 namespace Laminas\Cli\Input;
 
-use ArrayObject;
-use InvalidArgumentException;
-use RuntimeException;
-use Symfony\Component\Console\Application;
-use Symfony\Component\Console\Helper\QuestionHelper;
-use Symfony\Component\Console\Input\InputInterface;
-use Symfony\Component\Console\Input\InputOption;
-use Symfony\Component\Console\Output\OutputInterface;
-
-use function sprintf;
-
+/**
+ * Provide the majority of methods needed to implement InputParamInterface.
+ *
+ * This trait provides definitions for all but the following methods of the
+ * InputParamInterface:
+ *
+ * - getOptionMode()
+ * - getQuestion()
+ *
+ * Additionally, it defines the `$name` property, allowing implementations to
+ * set it in their constructors without needing to define the property
+ * themselves.
+ */
 trait InputParamTrait
 {
-    /**
-     * @internal
-     * @var null|ArrayObject<string, InputParam>
-     */
-    private $inputParams;
+    /** @var mixed */
+    private $default;
+
+    /** @var string */
+    private $description = '';
 
     /**
-     * @param null|mixed $default
-     * @return $this
-     * @throws RuntimeException
+     * Parameter name; must be set by class composing trait!
+     *
+     * @var string
      */
-    final public function addParam(
-        string $name,
-        string $description,
-        string $type,
-        bool $required = false,
-        $default = null,
-        array $options = []
-    ) : self {
-        $mode = $type === InputParam::TYPE_BOOL
-            ? InputOption::VALUE_NONE
-            : InputOption::VALUE_REQUIRED;
-
-        $this->addOption(
-            $name,
-            null,
-            $mode,
-            $description
-            // default null, on purpose
-        );
-
-        if ($this->inputParams === null) {
-            $this->inputParams = new ArrayObject();
-        } elseif (! $this->inputParams instanceof ArrayObject) {
-            throw new RuntimeException(sprintf(
-                'Command %s uses $inputParams property. It is not allowed while using %s',
-                static::class,
-                InputParamTrait::class
-            ));
-        }
-
-        $this->inputParams->offsetSet($name, new InputParam($name, $description, $type, $required, $default, $options));
+    private $name;
 
-        return $this;
-    }
+    /** @var bool */
+    private $required = false;
+
+    /** @var null|string */
+    private $shortcut;
 
     /**
-     * @return null|bool|int|string
-     * @throws InvalidArgumentException
+     * Default value to use if none provided.
+     *
+     * @return null|mixed
```

Comment:
I see on the property just `mixed` so I believe we should go the PHP 8.0 way, where mixed includes `null`.